pennylane/queuing.py

```diff
@@ -11,8 +11,115 @@
+within a ``with`` block.
+
+Let's take a look at an example. If we query the :class:`~.QueuingManager` outside of an
+:class:`~.AnnotatedQueue`'s context, we can see that nothing is recording and no active context exists.
+
+>>> print("Are we recording? ", qml.QueuingManager.recording())
+Are we recording?  False
+>>> print("What's the active context? ", qml.QueuingManager.active_context())
+What's the active context?  None
+
+Inside of a context, we can see the active recording context:
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     print("Are we recording? ", qml.QueuingManager.recording())
+...     print("Is q the active queue? ", q is qml.QueuingManager.active_context())
+Are we recording?  True
+Is q the active queue?  True
+
+If we construct an operator inside the recording context, we can see it is added to the queue:
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op = qml.PauliX(0)
+>>> q.queue
+[PauliX(wires=[0])]
+
+If an operator is constructed outside of the context, we can manually add it to the queue by
+calling the :meth:`~.Operator.queue` method:
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+>>> q.queue
+[PauliX(wires=[0])]
+
+An object can only exist up to *once* in the queue, so calling queue multiple times will
+not do anything.
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+...     op.queue()
+>>> q.queue
+[PauliX(wires=[0])]
+
+Instead, we need to queue a copy of the operator:
+
+>>> from copy import copy
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     op.queue()
+...     copy(op).queue()
+>>> q.queue
+[PauliX(wires=[0]), PauliX(wires=[0])]
+
+This can instead be accomplished with the :func:`~.apply` function:
+
+>>> op = qml.PauliX(0)
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     qml.apply(op)
+...     qml.apply(op)
+>>> q.queue
+[PauliX(wires=[0]), PauliX(wires=[0])]
+
+In the case of operators composed of other operators, each operator in the composite will be queued.
+For example, both ``PauliX`` and ``PauliX`` raised to a power.
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     base = qml.PauliX(0)
+...     pow_op = base ** 1.5
+>>> q.queue
+[PauliX(wires=[0]), PauliX(wires=[0])**1.5]
+
+In this case, each object will have metadata associated with it. At later processing steps, the original
+``PauliX`` will be ignored in favor of the operator that *owns* it.
+
+>>> q.get_info(base)
+{'owner': PauliX(wires=[0])**1.5}
+>>> q.get_info(base)["owner"] is pow_op
+True
+
+In order to construct new operators within a recording, but without queuing them, either
+use the :meth:`~.QueuingManager.stop_recording` context or specify `do_queue=False` upon construction:
+
+>>> with qml.queuing.AnnotatedQueue() as q:
+...     qml.PauliX(0, do_queue=False)
+...     with qml.QueuingManager.stop_recording():
+...         qml.PauliY(1)
+>>> q.queue
+[]
+
```

Comment:
Would it be too out of context to include something about how the queue is used? I guess not in-depth, since its not directly relevant for the functions here, but maybe just a reference to what uses the queue, so you know where to find that information if interested. 

I feel like this makes it very clear how the queue is constructed, but doesn't help me know where to go to find out what happens next.

Reply:
Added some information on `process_queue` and quantum script at the end.